.github/workflows/ci.yml

```diff
@@ -75,14 +75,14 @@ jobs:
       - uses: actions/checkout@v2
         with:
           submodules: "recursive"
-      - name: Build Software
-        run: |
-          make -C target/snitch_cluster sw \
-          CFG_OVERRIDE=cfg/snax-mac.hjson
       - name: Generate RTL
         run: |
           make -C target/snitch_cluster rtl-gen \
           CFG_OVERRIDE=cfg/snax-mac.hjson
+      - name: Build Software
+        run: |
+          make -C target/snitch_cluster sw \
+          CFG_OVERRIDE=cfg/snax-mac.hjson
```

Comment:
ah, because of the auto-generated header files?

Reply:
Yes 😢 so because every change in the streamer config generates a separate header file. We initially thought it should've been separated like separate rtl generation and header file generation.

However, @xiaoling-yi pointed out that it is more prone to errors if we have to separate them. Since every change in CSR config to generate RTL results in a change in streamer CSR as well.